### DISCUSSION TITLE
Core team sync 2024-04-22

### DISCUSSION THREAD

**Discussion by madolson** [valkey-io/valkey#362] [General]
1. Valkey summit
    1. Logistics (Omit? Maybe need call to action)
        1. June 5-6 in seattle is OK.
        2. @madolson to create an agenda to help motivate others to come
        3. current/future committers and users
        4. Attendance ommitted for privacy
        5. Goal is to get folks to attend in person. Will provide zoom links.
    2. Content
        1. Everyone presenting wishlist
        2. Everyone who has something to contribute
        3. Get alignments across contributors
        4. Have an agenda/specific topics
    2. Multi-year plan
        1. Future plan is to have annual Valkey conferences along with KubeCon/Linux OSS
1. Standalone Valkey redirect
    1. "Redirect" vs "Moved"
        2. Still unclear if we want multi-node support in standalone. How do clients handle a new node?
    3. Ready to merge?
        1. Recommendation is discussing in-person at Vakey summit (June)
        2. 
1. Clients
    1. Fork or not to fork
        1. Official forks under Valkey
        2. We should be proactive, not reactive
    2. Should we support more than one per language
        3. Initially just support one. Include more as needed.
    4. Should the languages be only for Valkey
        5. They should be generic RESP clients that support other implementations (Like KeyDB, Dragonfly)
        6. They shoulbe Valkey branded
6. Transparency
    7. How to invite other folks to meetings
        8. Setup a procedure for folks to indicate they want to bring up a topic. Details TBD.

**Comment by zuiderkwast** [comment]
Regarding forking clients, we agreed that to fork a client under valkey-io, we need someone to be an active maintainer of the forked client. It needs to be open source. Most other details are up to the client maintainer.

**Comment by virtualirfan** [comment]
Would like to attend summit to discuss a couple of proposals. Any idea yet on when will details be announced and where to send proposals/presentation abstracts for attendance qualification?

**Reply by madolson** [reply]
For lack of a better process, @virtualirfan would you send an email to me at maintainers@lists.valkey.io with your interest and what you would like to contribute.

**Comment by szuecs** [comment]
If you have a decision and it's pro forking clients into this org, I think it would be great for the Go community to fork https://github.com/redis/go-redis . It only recently moved to the redis org, but I expect the user base to be the biggest. We did some analysis last year if we want to change the client for https://github.com/zalando/skipper , but that seemed to be less attractive to us.

I tried to replace our redis setup with valkey and go-redis did not work. I did not see the PING command working with valkey commit 4948d536b91f23b9bdd505b1d356a063b06a8cd4. Not sure who breaks it. I was able to do a port-forward and connect to valkey via redis-cli, so I expect that go-redis is somehow wrong.
If you need help to make the first steps I am happy to help, but I can't really promise to maintain the project in the long run.

**Reply by murphyjacob4** [reply]
@szuecs Could you file a more detailed issue for the go-redis compatibility with Valkey? I am curious why this isn't working

**Reply by zuiderkwast** [reply]
We already have a valkey-go client. It's a fork of rueidis. Have you seen it? Is it worse than go-redis because it has fewer users or is there another reason?

**Reply by szuecs** [reply]
@murphyjacob4 I didn't figure out, yet. I will try to check it again next week. Basically PING via go-redis doesn't work. I am thinking right now that I may have a configuration error because maybe there's a redis.conf which normally configures the listener to bind on 0.0.0.0 and after valkey change it binds to localhost. I have to check it and report back.

@zuiderkwast we invested into auto scaling capability and use go-redis since some years. I will check the other, but I expect that go-redis has a much bigger user base. I checked the code and to switch we would need a couple of more Z commands, a ring client and a way to set the addresses dynamically of the redis instances in the ring (auto scaling). I am happy to help to get this in, if you are more happy with the chosen code than with the go-redis code. The code coverage seems to show the code is in a good shape. I didn't look into it deep enough to judge it.

**Reply by szuecs** [reply]
@murphyjacob4 
My bad `--protected-mode no` solved the problem. Valkey-server works as expected.

**Comment by Virusuki** [comment]
I would like to actively support and contribute to valkey to maintain the open source ecosystem.

I would like to continuously contribute to valkey open source as follows.
1) valkey code
2) valkey doc

We will think about valkey’s future and better processes and let you know.

I would like to attend the Valkey summit (June 5-6) via Zoom.
Are there any attendance requirements?

**Comment by tessus** [comment]
You should probably either unpin this completely or pin always the latest core team sync. What do you think?

**Reply by zuiderkwast** [reply]
We started using the wiki for meeting notes. I made a stub post with a link to that. Pinned that instead.
